cli/src/pcluster/cli/commands/pcluster3_configuration_converter.py

```diff
@@ -0,0 +1,593 @@
+        s3_read_write_resource = self.config_parser.get(
+            self.cluster_section_name, "s3_read_write_resource", fallback=None
+        )
+        if s3_read_resource:
+            bucket_name, key = get_bucket_name_and_key(s3_read_resource)
+            s3_access.append({"BucketName": bucket_name, "KeyName": key})
+        if s3_read_write_resource:
+            s3_access.append({"EnableWriteAccess": True, "BucketName": bucket_name, "KeyName": key})
+        if s3_access:
+            iam["S3Access"] = s3_access
+        if iam:
+            headnode["Iam"] = iam
+
+    def error(self, message):
+        """Raise SystemExit exception to the stderr."""
+        sys.exit("ERROR: {0}".format(message))
+
+    def convert_scheduling(self, section):
+        scheduling = {}
+        self.convert_single_field(self.cluster_section_name, "scheduler", scheduling, "Scheduler")
+        scheduler = self.config_parser.get(self.cluster_section_name, "scheduler", fallback=None)
+        if scheduler == "slurm":
+            self.convert_slurm_queue()
+        elif scheduler == "awsbatch":
+            pass
+
+        if scheduling:
+            self.pcluster3_configuration[section] = scheduling
+
+def _err_and_exit(message):
+    """Log the given error message and exit nonzero."""
+    print(message)
+    sys.exit(1)
+
+
+def _parse_args(argv=None):
+    """Parse command line args."""
+
+    parser = argparse.ArgumentParser(
+        description=("Updates the AWS ParallelCluster configuration file."),
+        epilog='For command specific flags, please run: "pcluster-config [command] --help"',
+    )
+    subparsers = parser.add_subparsers()
+    subparsers.required = True
+    subparsers.dest = "command"
+
+    convert_parser = subparsers.add_parser(
+        "convert",
+        help=(
+            "Convert a 'cluster' section from a ParallelCluster's configuration file "
+            "into a format that supports multiple instance types."
+        ),
+    )
+    convert_parser.add_argument(
+        "-c",
+        "--config-file",
+        help="Configuration file to be used as input.",
+    )
+    convert_parser.add_argument(
+        "-t",
+        "--cluster-template",
+        help=(
+            "Indicates the 'cluster' section of the configuration file to convert. "
+            "If not specified the script will look for the cluster_template parameter in the [global] section "
+            "or will search for '[cluster default]'."
+        ),
+        required=False,
+    )
+    convert_parser.add_argument(
+        "-o",
+        "--output-file",
+        help="Configuration file to be written as output. By default the output will be written to the stdout.",
+        required=True,
+    )
+    convert_parser.set_defaults(func=convert)
+
+    return parser.parse_args(argv)
+
+
+def convert(args=None):
+    try:
+        pcluster3_config_converter = Pcluster3ConfigConverter(args.config_file, args.cluster_template, args.output_file)
```

Comment:
`pcluster3_config_convert` feels verbose here -- could use `converter` for a local variable (since there are no other types of converters possible)

Reply:
Changed